src/main/java/com/algorithmtron/service/SortCharsByFrequencyService.java

```diff
@@ -0,0 +1,57 @@
+package com.algorithmtron.service;
+
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+public class SortCharsByFrequencyService {
+
+    public String frequencySort(String s) {
+        Map<Character, Integer> hm = new HashMap<>();
+        StringBuilder sb = new StringBuilder();
+
+        for (char c : s.toCharArray()) {
+            if (!hm.containsKey(c)) {
+                hm.put(c, 1);
+            } else {
+                hm.put(c, hm.get(c) + 1);
+            }
+        }
+        int max = Integer.MIN_VALUE;
```

Comment:
Maybe we could move this max value somewhere outside the function, since it is a global constant value.

Also rename it to MAX, MAX_value or similar. Constants such as this one should be upper case with "_" between the words.

Reply:
Nevermind, we keep reinitializing it, not a constant